includes/admin/class-wp-job-manager-promoted-jobs.php

```diff
@@ -0,0 +1,183 @@
+	}
+
+	/**
+	 * Add a column to the job listings admin page.
+	 *
+	 * @param array $columns Columns.
+	 * @return array
+	 */
+	public function columns( $columns ) {
+		$columns['promoted_jobs'] = __( 'Promote', 'wp-job-manager' );
+		return $columns;
+	}
+
+	/**
+	 * Check if a job is promoted.
+	 *
+	 * @param [string] $post_id
+	 * @return boolean
+	 */
+	public function is_promoted( $post_id ) {
+		$promoted = get_post_meta( $post_id, '_promoted', true );
+		return $promoted ? true : false;
+	}
+
+	/**
+	 * Handle display of new column
+	 *
+	 * @param  string $column
+	 */
+	public function custom_columns( $column ) {
+		global $post;
+		if ( 'promoted_jobs' === $column ) {
+			if ( $this->is_promoted( $post->ID ) ) {
+				echo '
+					Live
+					<br />
+					<a href="#">Edit promotion</a>
+					<br />
+					<a href="#">Deactivate</a>
+				';
+			} else {
+				echo '<button class="promote_job button button-primary" data-post=' . esc_attr( $post->ID ) . '>Promote</button>';
+			}
+		}
+	}
+	/**
+	 * Store the promoted jobs template from wpjobmanager.com
+	 *
+	 * TODO: we need to fetch this from wpjobmanager.com and store in cache for X amount of time
+	 * We should also have a fallback in case the API call fails.
+	 *
+	 * @return string
+	 */
+	public function get_promote_jobs_template() {
+		return '
+		<dialog id="promoteDialog">
+			<form method="dialog">
+				<button type="submit" autofocus>X</button>
+			</form>
+			<promote-job-modal>
+				<div slot="column-left" class="column-left">
+					<h2 slot="promote-heading">
+					Promote yoor job on our partner network.
+					</h2>
+
+					<div slot="price" class="price">
+						Starting From
+						<span>$83.00</span>
+					</div>
+
+					<div slot="promote-list">
+						<ul>
+							<li>Your ad will get shared on our Partner Network</li>
+							<li>Featured on jobs.blog for 7 days</li>
+							<li>Featured on our weekly email blast</li>
+						</ul>
+					</div>
+
+					<div slot="buttons" class="promote-buttons-group">
+						<button class="button button-primary" type="submit">Promote your job</button>
+						<button class="button button-secondary" type="submit">Learn More</button>
+					</div>
+				</div>
+				<div slot="column-right" class="column-right">
+					<img src="https://d.pr/i/4PgTqN+">
+				</div>
+			</promote-job-modal>
+		</dialog>';
+	}
+
+	/**
+	 * Output the promote jobs template
+	 *
+	 * @return void
+	 */
+	public function my_admin_notice() {
+		echo '
+			<template id="promote-job-template">
+			<style>
+
+			</style>
+				<slot name="column-left" class="column-left">
+					<slot name="promote-heading">
+					Promote Your Job on our Partner Network
+					</slot>
+
+					<slot name="price">
+						Starting from
+						<span>$80.00</span>
+					</slot>
+
+					<slot name="promote-list">
+						<ul>
+							<li>Your ad will get shared on our Partner Network</li>
+							<li>Featured on jobs.blog for 7 days</li>
```

Comment:
Since it's a default content that could be used in some situation, maybe we should only use more generic information?
If it makes sense, we would need to be a little creative here to write some stuff that will never change, like _"Promote your job on external job boards"_.
It would also apply to the price, maybe having something like "$ --".

Reply:
So one weird caveat I found is that you can't have a `dialog` inside of a `template` - because of how the Shadow DOM works and how components are mounted.  So we need to have a fallback in `get_promote_jobs_template` that has the dialog in it.  Just wanted to make a note if it here!